object-store-client-common/src/main/scala/uk/gov/hmrc/objectstore/client/ObjectStoreClient.scala

```diff
@@ -74,7 +74,7 @@ class ObjectStoreClient[F[_], REQ_BODY, RES, RES_BODY](
     path: Path.Directory,
     owner: String = config.owner
   ): F[ObjectListing] = {
-    val effectivePath = if (path.asUri == "/") "" else s"${path.asUri}"
+    val effectivePath = if (path.asUri.isEmpty) "" else s"${path.asUri}/"
     F.flatMap(client.get(s"$url/list/$owner/$effectivePath", List(authorizationHeader)))(read.toObjectListing)
```

Comment:
why can't you not just do `s"$url/list/$owner/${path.asUri}"` ?

Reply:
We need a trailing `/` in the final path. The s3 key generated from this path will need a trailing `/` to list all objects under that path.